### PR TITLE
[Serve][Doc] Add notes about def endpoint in FastAPI

### PR DESCRIPTION
**Pull request by sihanwang41**
Signed-off-by: Sihan Wang <sihanwang41@gmail.com>

## Why are these changes needed?

Add a note to indicate the def endpoint potential issue in ray serve.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/performance.md [hunk 1]
```diff
@@ -150,6 +150,10 @@ There are handful of ways to address these issues:
 (serve-performance-async-methods)=
 ### Using `async` methods
 
+:::{note}
+According to the [FastAPI documentation](https://fastapi.tiangolo.com/async/#very-technical-details), `def` endpoint function will be called in a separate threadpool, you might observe many requests running at the same time inside one replica, and this scenario might cause the OOM or resource starvation. In this case, you can try to use `async def` and tune `max_concurrent_queries` to control the workload performance.
```
Comment (architkulkarni) [inline doc/source/serve/performance.md:154]:
```suggestion
According to the [FastAPI documentation](https://fastapi.tiangolo.com/async/#very-technical-details), `def` endpoint functions will be called in a separate threadpool, so you might observe many requests running at the same time inside one replica, and this scenario might cause OOM or resource starvation. In this case, you can try to use `async def` and tune `max_concurrent_queries` to control the workload performance.
```
By "tune" do we just mean increase? (i.e. if the user sees OOM/resource starvation -> user should increase max_concurrent_queries). If so let's just say "increase".

Comment (sihanwang41) [reply doc/source/serve/performance.md:154]: Directly remove `max_concurrent_queries` for now, it is mentioned in the following section clearly.

Comment (shrekris-anyscale) [reply doc/source/serve/performance.md]: Why are we removing `max_concurrent_queries`?

Comment (sihanwang41) [reply doc/source/serve/performance.md]: it is mentioned the `max_concurrent_queries` in the following section. (redundant info)

Comment (sihanwang41) [reply doc/source/serve/performance.md]: Not directly related to the `def` and `async def` information

**Review by shrekris-anyscale — APPROVED** [review]

**Comment by architkulkarni** [comment]
Linkcheck failures are from RLlib, unrelated

**Pull request merged**